src/predictor/gpu_predictor.cu

```diff
@@ -361,10 +364,14 @@ class GPUPredictor : public xgboost::Predictor {
       DeviceOffsets(batch.offset, batch.data.Size(), &device_offsets);
       batch.data.Reshard(GPUDistribution::Explicit(devices_, device_offsets));
 
-      // TODO(rongou): only copy the model once for all the batches.
+      if (batch_offset == 0) {
+        dh::ExecuteIndexShards(&shards_, [&](int idx, DeviceShard& shard) {
```

Comment:
Given that the code processing the tree nodes above (lines 335-350) has to do with model initialization, consider moving it (together with calls to `DeviceShard::InitModel()`) to a separate method of GPUPredictor.

Reply:
Done.